src/PDPVerifier.sol

```diff
@@ -143,6 +143,14 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         challengeFinality = _challengeFinality;
     }
 
+    string public constant VERSION = "1.1.0";
+    event ContractUpgraded(string version, address implementation);
+
+    function migrate() public onlyProxy reinitializer(2) {
+        require(msg.sender == address(this), "Only callable by self during upgrade");
```

Comment:
Main thing I am still not able to understand is how this works
        require(msg.sender == address(this), "Only callable by self during upgrade");
How can msg.sender  by the same as the proxy addresses if migrate  is essentially called through a delegatecall by the proxy as part of upgradeToAndCall  ?
Shouldn't msg.sender be the EOA that triggered the upgrade ?